# HG changeset patch
# User Takuro Ashie <ashie@clear-code.com>
# Date 1536566532 -32400
#      Mon Sep 10 17:02:12 2018 +0900
# Node ID 119260545b752ef1c7ca54afa7c94a3b090b8199
# Parent  9bb0f7fc73ad2109e4ea777e9ba65a16a6acc9cd
Plug memory leak of PureOmxPlatformLayer

diff --git a/dom/media/platforms/omx/PureOmxPlatformLayer.cpp b/dom/media/platforms/omx/PureOmxPlatformLayer.cpp
--- a/dom/media/platforms/omx/PureOmxPlatformLayer.cpp
+++ b/dom/media/platforms/omx/PureOmxPlatformLayer.cpp
@@ -111,19 +111,16 @@ PureOmxPlatformLayer::PureOmxPlatformLay
   , mImageContainer(aImageContainer)
 {
   LOG("");
 }
 
 PureOmxPlatformLayer::~PureOmxPlatformLayer()
 {
   LOG("");
-  if (mComponent) {
-    OMX_FreeHandle(mComponent);
-  }
 }
 
 OMX_ERRORTYPE
 PureOmxPlatformLayer::InitOmxToStateLoaded(const TrackInfo* aInfo)
 {
   LOG("");
 
   if (!aInfo) {
@@ -272,16 +269,22 @@ PureOmxPlatformLayer::SetParameter(OMX_I
                           aParamIndex,
                           aComponentParameterStructure);
 }
 
 nsresult
 PureOmxPlatformLayer::Shutdown()
 {
   LOG("");
+  if (mComponent) {
+    OMX_FreeHandle(mComponent);
+  }
+  mComponent = nullptr;
+  mPromiseLayer = nullptr;
+  mDataDecoder = nullptr;
   return NS_OK;
 }
 
 /* static */ OMX_ERRORTYPE
 PureOmxPlatformLayer::EventHandler(OMX_HANDLETYPE hComponent,
                                    OMX_PTR pAppData,
                                    OMX_EVENTTYPE eEventType,
                                    OMX_U32 nData1,
